src/CheckstyleCodeActions.hx

```diff
@@ -0,0 +1,91 @@
+import haxe.extern.EitherType;
+import vscode.CancellationToken;
+import vscode.CodeAction;
+import vscode.CodeActionContext;
+import vscode.CodeActionKind;
+import vscode.Command;
+import vscode.Diagnostic;
+import vscode.Position;
+import vscode.ProviderResult;
+import vscode.Range;
+import vscode.TextDocument;
+import vscode.WorkspaceEdit;
+
+class CheckstyleCodeActions {
+
+    public function new() {}
+
+    public function provideCodeActions(document:TextDocument, range:Range, context:CodeActionContext, token:CancellationToken):ProviderResult<Array<EitherType<Command,CodeAction>>> {
+        var commands:Array<EitherType<Command,CodeAction>> = [];
+        for (diag in context.diagnostics) {
+            if (diag.source != "checkstyle") {
+                continue;
+            }
+            var index = diag.message.indexOf(" - ");
+            if (index <= 0) {
+                continue;
+            }
+            var checkName:String = diag.message.substr(0, index);
+            switch (checkName) {
+                case "Dynamic":
```

Comment:
If more of these are added, it would be nice to have an enum abstract for these eventually to avoid stringly typed code.